tasks/install.yml

```diff
@@ -42,7 +42,7 @@
     owner: "{{ tomcat_user }}"
     group: "{{ tomcat_group }}"
     mode: 0750
-  when: 'tomcat_force_reinstall or tomcat_check is failed or tomcat_check.stdout == "Apache Tomcat/tomcat_version"'
+  when: 'tomcat_force_reinstall or tomcat_check is failed or tomcat_check.stdout != "Apache Tomcat/{{ tomcat_version }}"'
```

Comment:
It seems that the linter doesn't like jinja2 templating inside when conditionals.
From travis builds running ansible-lint: `tasks/install.yml:36: [E102] No Jinja2 in when`

As an alternative you may like to consider something like `or tomcat_version not in tomcat_check.stderr` (Notice there is no quotations here)

Reply:
I did some basic task tests with the `not in` conditional to see if it would work:

playbook:

```
- name: test
  hosts: all
  connection: local
  vars:
    tomcat_version_same: 8.0.0
    tomcat_version_diff: 9.3.14
    tomcat_check:
      stdout: "Apache Tomcat/8.0.0"


  tasks:
    - name: Comparison
      debug: msg="Same version detected"
      when: tomcat_version_same in tomcat_check.stdout

    - name: Comparison not
      debug: msg="Different version detected"
      when: tomcat_version_diff not in tomcat_check.stdout

```

Execution:

```
$ ansible-playbook -i localhost, play.yml 
[...]
ok: [localhost]

TASK [Comparison] ****************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "Same version detected"
}

TASK [Comparison not] ************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "Different version detected"
}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

Seems to work fine. I pushed a commit to fix it.

Thanks!